activerecord/lib/active_record/counter_cache.rb

```diff
@@ -73,13 +88,34 @@ def reset_counters(id, *counters)
       #   # UPDATE posts
       #   #    SET comment_count = COALESCE(comment_count, 0) + 1
       #   #  WHERE id IN (10, 15)
+      #
+      #   # For the Posts with id of 10 and 15, increment the comment_count by 1
+      #   # and update the updated_at value for each counter.
+      #   Post.update_counters [10, 15], comment_count: 1, touch: true
+      #   # Executes the following SQL:
+      #   # UPDATE posts
+      #   #    SET comment_count = COALESCE(comment_count, 0) + 1,
+      #   #    `updated_at` = '2016-10-13T09:59:23-05:00'
+      #   #  WHERE id IN (10, 15)
       def update_counters(id, counters)
+        if touch = counters.delete(:touch)
+          object = find(id)
+          touch_time = object.send(:current_time_from_proper_timezone)
+          timestamps = resolve_timestamp_columns(object, touch)
+        else
+          timestamps = []
+        end
+
```

Comment:
Think it's odd we split out the timestamp handling to two places. Consider moving it like so:

```ruby
touch = counters.delete(:touch)

updates = counters.map # …

if touch
  object = find(id)
  touch_time = object.send(:current_time_from_proper_timezone)
  resolve_timestamp_columns(object, touch).each do |column|
    updates << "#{connection.quote_column_name(column.to_s)} = '#{touch_time}'"
  end
end
```

Spares the local variable and also fixes the `timestamps.map` to use `each` :)